SU-SWS/ace-sulgryphon
# SUP-63: Book content type

# READY FOR REVIEW

# Summary
- Added content type for books, migration, and configuration for pathauto and apis, as well as taxonomies to support the imported data
- The migrations will import all the fields with the exception of images.  We are still waiting on images to get pulled in from the API (should be done in a few more days).  The image field is already in the book record, so when that data is available, we can update the migration using the instructions [here](https://github.com/SU-SWS/ace-stanfordlagunita/pull/80) to link the book covers with the correct books.

# Review By (Date)
- When does this need to be reviewed by?

# Urgency
- How critical is this PR?

# Steps to Test

1. Check out the branch
2. `blt drupal:update --site=supress`
3.  Add to your `local.settings.php` for supress:
```
$config['migrate_plus.migration.sup_import_books']['source']['authentication'] = [
  'token_url' => 'https://memento.stanford.edu/fmi/data/v2/databases/Web/sessions',
  'client_id' => '[your api username]',
  'client_secret' => '[your api password]',
];
```
4.  Check to see the content type `Book` exists.  Check for the `Book Subjects`, `Imprints` and `Series` taxonomies.
5. Visit `/admin/structure/migrate/manage/su_press/migrations`.  You should see two migrations listed.
6. Run the `Import Awards` migration first, since the book importer uses that information in entity lookups.
7. run the `Import Stanford University Press Books` migration.  This will  take some time to complete, and it will only grab the first 1000 books.
8. You will want to run the same migration a second time, with the `Update` option checked.  That will allow it to update related books.  (it won't find all of them, since it's only the first 1000 books for convenience).
9. Go to `/admin/content`, and search for `falconry`.  That should return you one book.  Edit the record, and confirm that there is a related title.
10. Try anything else you like and make sure I haven't missed anything.
# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
